infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -421,25 +423,39 @@ private synchronized void runDetectionTasks() {
      * @return Detector task.
      */
     private DetectorTask restoreRedundancyAndMergeSegments(Layout layout) {
-        int segmentsCount = layout.getSegments().size();
+        String localEndpoint = serverContext.getLocalEndpoint();
 
-        if (segmentsCount == 1) {
-            log.debug("No segments to merge. Skipping step.");
-            return DetectorTask.SKIPPED;
+        if (RedundancyCalculator.requiresMerge(layout, localEndpoint) &&
```

Comment:
Add comments to explain this check. It is not very intuitive.

Reply:
Sounds good